feat(curriculum): Build a Zoo Animal Scatterplot (#66379)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-build-a-zoo-animal-scatterplot/69bbd3f66dfa39e277252772.md

---
id: 69bbd3f66dfa39e277252772
title: Step 10
challengeType: 0
dashedName: step-10
---

# --description--

Now, you will learn about scale generator of d3 library.   
There several types of [scale generators](https://d3js.org/d3-scale) in D3 library. Here, you will focus in two scale generator: 

  1. scaleOrdinal()

>   2. scaleLinear()

Reviewer: This can be removed. And normally we don't include links in the workshop steps.

> In this step, you will learn about `scaleOrdinal()`.

Reviewer: ```suggestion
In this step, you will create a category to color mapping function using the `scaleOrdinal` function.
`scaleOrdinal` maps categories to corresponding values like colors or labels. In this case, you will map animal categories to color values.
```

> *scaleOrdinal()*

Reviewer: ```suggestion
Here is an example:
```

```js
const color = d3.scaleOrdinal(["color1", "color2", "color3"], ["red", "green", "blue"]);
color("color1") // "red"
```

All element first array `["color1", "color2", "color3"]` of ordinal scale get mapped with each element of second array `["red", "green", "blue"]`.   

> Now, create a ordinal scale for type to color named as `color`. ["Mammal", "Bird", "Reptile"] -> ["red", "green", "blue"]

Reviewer: ```suggestion
All elements of the first array `["color1", "color2", "color3"]` get mapped with each element of the second array `["red", "green", "blue"]`.   

Create an ordinal scale named `color` that maps the animal types `["Mammal", "Bird", "Reptile"]` to the colors `["red", "green", "blue"]`.
```

# --hints--

> You should create ordinal scale as `color`.

Reviewer: ```suggestion
You should create a variable called `color` and assign it `d3.scaleOrdinal()`.
```
